File: prayer_app_flutter/lib/src/screens/demo_screen.dart
import 'package:flutter/material.dart';
import 'package:material_design_icons_flutter/material_design_icons_flutter.dart';
import '../theme/app_theme.dart';
import '../components/screen_container.dart';
import '../components/glass_card.dart';
import '../components/app_header.dart';
import '../components/app_icon_button.dart';
import '../components/app_divider.dart';
import '../components/bottom_nav_bar.dart';

class DemoScreen extends StatelessWidget {
  const DemoScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: ScreenContainer(
        child: ListView(
          padding: const EdgeInsets.all(AppSpacing.s16),
          children: [
            const AppHeader(title: 'Bucharest'),
            const SizedBox(height: AppSpacing.s24),

            // Typography
            Text('Title Large', style: AppTypography.titleLarge),
            const SizedBox(height: AppSpacing.s8),
            Text('Title Medium', style: AppTypography.titleMedium),
            const SizedBox(height: AppSpacing.s8),
            Text('Body text looks like this', style: AppTypography.body),
            const SizedBox(height: AppSpacing.s8),
            Text('Caption text is muted', style: AppTypography.caption),

            const SizedBox(height: AppSpacing.s24),
            const AppDivider(),
            const SizedBox(height: AppSpacing.s24),

            // Glass Card
            GlassCard(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text('Next Prayer: MAGHRIB', style: AppTypography.body),
                  const SizedBox(height: AppSpacing.s8),
                  Text('Starts in 02:14:30', style: AppTypography.titleLarge),
                  const SizedBox(height: AppSpacing.s4),
                  Text('Adhan at 5:50 PM', style: AppTypography.caption),
                ],
              ),
            ),

            const SizedBox(height: AppSpacing.s24),

            // Button row
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceEvenly,
              children: [
                AppIconButton(
                  icon: MdiIcons.cogOutline,
                  onTap: () {},
                ),
                AppIconButton(
                  icon: MdiIcons.bellOutline,
                  onTap: () {},
                ),
                AppIconButton(
                  icon: MdiIcons.shareVariant,
                  onTap: () {},
                ),
              ],
            ),

            const SizedBox(height: AppSpacing.s24),

            // Color swatches
            _swatch('backgroundStart', AppColors.backgroundStart),
            _swatch('backgroundEnd', AppColors.backgroundEnd),
            _swatch('card', AppColors.card),
            _swatch('accentGold', AppColors.accentGold),
            _swatch('textMuted', AppColors.textMuted),
            _swatch('inactive', AppColors.inactive),
          ],
        ),
      ),
    );
  }

  Widget _swatch(String label, Color color) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: AppSpacing.s4),
      child: Row(
        children: [
          Container(
            width: 32,
            height: 32,
            decoration: BoxDecoration(
              color: color,
              borderRadius: BorderRadius.circular(8),
              border: Border.all(color: AppColors.cardBorder),
            ),
          ),
          const SizedBox(width: AppSpacing.s12),
          Text(label, style: AppTypography.body),
        ],
      ),
    );
  }
}

File: prayer_app_expo/src/screens/DemoScreen.js
import React from 'react';
import { View, Text, ScrollView, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { Colors, Spacing, Radius, Typography, interFont } from '../theme/theme';
import ScreenContainer from '../components/ScreenContainer';
import GlassCard from '../components/GlassCard';
import AppHeader from '../components/AppHeader';
import AppIconButton from '../components/AppIconButton';
import AppDivider from '../components/AppDivider';

export default function DemoScreen() {
    return (
        <ScreenContainer>
            <ScrollView style={styles.scroll} contentContainerStyle={styles.content}>
                <AppHeader title="Bucharest" />
                <View style={{ height: Spacing.s24 }} />

                {/* Typography */}
                <Text style={Typography.titleLarge}>Title Large</Text>
                <View style={{ height: Spacing.s8 }} />
                <Text style={Typography.titleMedium}>Title Medium</Text>
                <View style={{ height: Spacing.s8 }} />
                <Text style={Typography.body}>Body text looks like this</Text>
                <View style={{ height: Spacing.s8 }} />
                <Text style={Typography.caption}>Caption text is muted</Text>

                <View style={{ height: Spacing.s24 }} />
                <AppDivider />
                <View style={{ height: Spacing.s24 }} />

                {/* Glass Card */}
                <GlassCard>
                    <Text style={Typography.body}>Next Prayer: MAGHRIB</Text>
                    <View style={{ height: Spacing.s8 }} />
                    <Text style={Typography.titleLarge}>Starts in 02:14:30</Text>
                    <View style={{ height: Spacing.s4 }} />
                    <Text style={Typography.caption}>Adhan at 5:50 PM</Text>
                </GlassCard>

                <View style={{ height: Spacing.s24 }} />

                {/* Button row */}
                <View style={styles.buttonRow}>
                    <AppIconButton icon="cog-outline" onPress={() => { }} />
                    <AppIconButton icon="bell-outline" onPress={() => { }} />
                    <AppIconButton icon="share-variant" onPress={() => { }} />
                </View>

                <View style={{ height: Spacing.s24 }} />

                {/* Color swatches */}
                {[
                    { label: 'backgroundStart', color: Colors.backgroundStart },
                    { label: 'backgroundEnd', color: Colors.backgroundEnd },
                    { label: 'card', color: Colors.card },
                    { label: 'accentGold', color: Colors.accentGold },
                    { label: 'textMuted', color: Colors.textMuted },
                    { label: 'inactive', color: Colors.inactive },
                ].map((s) => (
                    <View key={s.label} style={styles.swatch}>
                        <View style={[styles.swatchBox, { backgroundColor: s.color }]} />
                        <Text style={Typography.body}>{s.label}</Text>
                    </View>
                ))}
            </ScrollView>
        </ScreenContainer>
    );
}

const styles = StyleSheet.create({
    scroll: { flex: 1 },
    content: { padding: Spacing.s16 },
    buttonRow: {
        flexDirection: 'row',
        justifyContent: 'space-evenly',
    },
    swatch: {
        flexDirection: 'row',
        alignItems: 'center',
        marginVertical: Spacing.s4,
    },
    swatchBox: {
        width: 32,
        height: 32,
        borderRadius: 8,
        borderWidth: 1,
        borderColor: Colors.cardBorder,
        marginRight: Spacing.s12,
    },
});
